### DISCUSSION TITLE
Custom fields in article layout builder

### DISCUSSION THREAD

**Discussion by vtheod** [templaza/astroid-framework#914] [Ideas]
Hello @sonvnn 

I was wondering if it could be possible to add custom fields in the layout builder. 
I believe that it would be a great functionality for the Pro version.
I tried to do it in the past but due to lack of free time I abandoned it.

Best regards,
Vassilis

**Comment by Banuk1616** [comment]
I agree 100%, this would be a fantastic feature for the Pro version.
Best regards

**Reply by rinka88** [reply]
should be in the plans, since dynamic content is indicated )

**Reply by Banuk1616** [reply]
You are right, but unfortunately it is not possible to output the contents of Custum Fields from different field groups at any position in an article. I started a question about this a long time ago.

https://github.com/templaza/astroid-framework/discussions/765

To achieve this goal anyway, I worked with overrides.

It would be a fantastic feature for Astroid if this were possible without overrides at some point. 
Best regards

**Comment by vtheod** [comment]
I came up with a solution. I created an element which can load custom fields or custom fields groups. I still test it though

**Reply by rinka88** [reply]
If you can interact with dynamic content, that would be great.

**Reply by Banuk1616** [reply]
> If you can interact with dynamic content, that would be great.

That would be great, please keep us updated.

**Reply by rinka88** [reply]
@vtheod How is this idea developing?
@pepperstreet you can follow this thread, it might help

**Reply by rinka88** [reply]
@sonvnn Is this functionality planned?

**Comment by rinka88** [comment]
@vtheod @sonvnn Good day. Is there any news on this topic?

**Reply by sonvnn** [reply]
This feature is on our to-do list. I will let you know as soon as it's released.

**Comment by vtheod** [comment]
@rinka88 @sonvnn @pepperstreet
Hello, I developed a custom fields element for the layout builder. You can add as many custom fields (or fields groups) as you want. 
It is developed as a Joomla plugin, so you can install it like any other extension. You can check the following video if you want:


https://github.com/user-attachments/assets/cfaeb82a-5207-4860-8a93-5603f2f63721

**Reply by rinka88** [reply]
This is a great widget solution, and it's definitely worth it.👍 If the solution is public and free, then it's probably worth putting it online (GitHub or elsewhere).

**Reply by rinka88** [reply]
But it would also be great in the form of dynamic content.

**Reply by rinka88** [reply]
@sonvnn Maybe we could create a section on the website for custom widgets from other developers. So that we could post widgets for Astroid that are different from the core, if we have our own.

**Reply by sonvnn** [reply]
@rinka88 @vtheod That’s awesome! I love that idea. I’ll set up a channel for users to upload their products, and create a page on the Astroid website to feature and link to their creative projects.

**Reply by sonvnn** [reply]
@vtheod Please create a topic to share your widget here. I will share it to Astroid Website and link to your download page. Please follow this rule https://github.com/templaza/astroid-framework/discussions/1324
Thanks & Best Regards,
Sonny

**Comment by rinka88** [comment]
@vtheod I apologize, but will there be any news or links regarding this decision?

**Reply by vtheod** [reply]
Sorry for my absence, I’m going through an extremely busy period right now. I believe that, at the latest, I’ll be able to manage it during the Christmas holidays.

**Reply by rinka88** [reply]
Hello. Happy New Year!

**Comment by rinka88** [comment]
@vtheod still no news?

**Comment by rinka88** [comment]
@vtheod Good day. I apologize, but is it possible to somehow test this solution?

**Reply by vtheod** [reply]
Hello @rinka88 
Due to the time and effort invested in development, the extension will not be free. The delay is because I'm upgrading my site so that I can also offer extensions for sale.

**Comment by jam1974** [comment]
Here you go: https://extensions.joomla.org/extension/oj-astroid-custom-content-fields/

**Comment by iorbita** [comment]
Hello,

Is there any chance of seeing native Joomla Custom Fields integration in Astroid Framework in the future? Even with the Pro version, I feel that without Custom Fields support, the current approach remains quite limited.

Specifically, I'm thinking about integration with Tassos' Advanced Custom Fields extension. If that could be integrated into Astroid natively, it would be a major step forward for the framework.

This issue is particularly noticeable in the "Media" section: having all those elements without the ability to manage media sources dynamically doesn't really make sense to me. Furthermore, under the "Advanced" tab for "Dynamic Content Settings" in Media elements, only articles, categories and users are available as options. What is the intended use case for Media content here? Am I missing something in the current workflow?

On a side note regarding Custom Fields: I noticed that Oliver Jentschel offers an extension for Astroid to bridge this gap. Does anyone know if Tassos' Advanced Custom Fields are supported/compatible with it? I sent a message through his website, but haven't received a reply yet.

Thanks!

**Comment by rinka88** [comment]
Yes, this is one of the long-awaited features for Astroid.🙂
 @sonvnn promised to add it in one of the upcoming releases.
